### PR TITLE
[WIP] Add a blocklist (currently empty)

### PR DESCRIPTION
**Pull request by PlasmaPower**
I think this might require additional user permissions for the access token.

### CODE REVIEW

**Comment by MCSH** [comment]
I don't think blocking based on login is a good idea (login could be changed people!!) use id instead.

**Comment by PlasmaPower** [comment]
@MCSH hmm, that currently isn't possible with this API. I'll just not unblock users I guess.

File: chaos.py [hunk 1]
```diff
@@ -50,6 +50,13 @@ def main():
     log.info("Setting description to {desc}".format(desc=settings.REPO_DESCRIPTION))
     github_api.repos.set_desc(api, settings.URN, settings.REPO_DESCRIPTION)
 
+    log.info("Blocking users specified in blocklist")
+    with open(join(dirname(__file__), "blocklist"), "r") as blocklist:
+        try:
+            github_api.users.set_blocklist(api, blocklist.readlines())
+        except:
+            log.exception("Failed to update blocklist")
```
Comment (amoffat) [inline chaos.py:58]: I'm into the idea of wrapping non-core functionality in good error handling :+1:

**Comment by amoffat** [comment]
[Blocking details](https://help.github.com/articles/blocking-a-user-from-your-personal-account/), just for documentation:

At the time that you block a user:

* The user stops following you
* The user stops watching your repositories
* The user's stars and issue assignments are removed from your repositories
* The user is removed as a collaborator on your repositories
* You're removed as a collaborator on their repositories

After you've blocked a user, they cannot:

* Send you any notifications, including by @mentioning your username
* Follow you or see your content in their activity feed
* Invite you as a collaborator on their repositories
* Cross-reference your repositories in comments
* Fork, watch, or star your repositories

In repositories you own, blocked users also cannot:

* Open issues or send pull requests
* Comment on issues, pull requests, or commits
* Add or edit wiki pages

File: chaos.py (hunk 1 above)
Comment (PlasmaPower) [reply chaos.py:58]: I'm pretty sure you understood this, but that doesn't count as a positive vote.

**Comment by PlasmaPower** [comment]
Marking as WIP until #343 reaches its conclusion

**Comment by yet-another-account** [comment]
I don't think we should block @smittyvb entirely, rather, lessen his voting power a bit and bump his voting threshold requirement a bit. After all, it's no fun if we ban him completely.

**Comment by phil-r** [comment]
I agree with @eukaryote31 on this one

**Comment by PlasmaPower** [comment]
@eukaryote31 for me at least, the number of PRs opened obviously trying to take over got annoying. I'll remove smittyvb from the blocklist if #343 turns out to be good. Would you vote for this PR if the blocklist was empty? (just a tool for future use)

**Comment by phil-r** [comment]
@PlasmaPower I thinks it's fine to have a list without smittyvb in it

**Comment by PlasmaPower** [comment]
Okay. I'll make that change now, then remove WIP after #343 is done.

**Comment by PlasmaPower** [comment]
I also undid the changes to `voting.py` (previously I had removed the 1/2 vote for smittyvb).

**Comment by phil-r** [comment]
@PlasmaPower You forgot to update title ;)

**Comment by PlasmaPower** [comment]
@phil-r done :)

Somewhat ironically, Travis is failing because of the hotfix that removed the check on Travis.

**Comment by phil-r** [comment]
@PlasmaPower but it's not required anymore, so it's okay (but it will be fixed in #337)

**Comment by PlasmaPower** [comment]
@phil-r Yeah I know, the hotfix that did that also broke Travis.

**Comment by PlasmaPower** [comment]
Okay #343 got through without a problem, and I still haven't collected the required votes, so I'm just going to wait until we actually need a blocklist to merge this.

**Pull request closed without merging**